File: sdk/samples/WCFSamples/TechnologySamples/Basic/Binding/MsmqIntegration/MessageCorrelation/CS/Service/OrderResponse.cs

//  Copyright (c) Microsoft Corporation.  All Rights Reserved.

using System.ServiceModel;
using System.ServiceModel.Channels;
using System.ServiceModel.MsmqIntegration;
using Microsoft.ServiceModel.Samples;

namespace Microsoft.ServiceModel.Samples
{
    [System.ServiceModel.ServiceContractAttribute(Namespace = "http://Microsoft.ServiceModel.Samples")]
    public interface IOrderResponse
    {

        [System.ServiceModel.OperationContractAttribute(IsOneWay = true, Action = "*")]
        void SendOrderResponse(MsmqMessage<PurchaseOrder> msg);
    }

    public partial class OrderResponseClient : System.ServiceModel.ClientBase<IOrderResponse>, IOrderResponse
    {

        public OrderResponseClient()
        { }

        public OrderResponseClient(string configurationName)
            : base(configurationName)
        { }

        public OrderResponseClient(System.ServiceModel.Channels.Binding binding, System.ServiceModel.EndpointAddress address)
            : base(binding, address)
        { }

        public void SendOrderResponse(MsmqMessage<PurchaseOrder> msg)
        {
            base.Channel.SendOrderResponse(msg);
        }
    }
}

File: sdk/samples/WCFSamples/TechnologySamples/Basic/Binding/MsmqIntegration/MessageCorrelation/VB/Service/OrderResponse.vb
' Copyright (c) Microsoft Corporation.  All Rights Reserved.

Imports System.ServiceModel
Imports System.ServiceModel.Channels
Imports System.ServiceModel.MsmqIntegration
Imports Microsoft.ServiceModel.Samples

Namespace Microsoft.ServiceModel.Samples

    <System.ServiceModel.ServiceContractAttribute([Namespace]:="http://Microsoft.ServiceModel.Samples")> _
    Public Interface IOrderResponse

        <System.ServiceModel.OperationContractAttribute(IsOneWay:=True, Action:="*")> _
        Sub SendOrderResponse(ByVal msg As MsmqMessage(Of PurchaseOrder))

    End Interface

    Partial Public Class OrderResponseClient
        Inherits System.ServiceModel.ClientBase(Of IOrderResponse)
        Implements IOrderResponse

        Public Sub New()

        End Sub

        Public Sub New(ByVal configurationName As String)

            MyBase.New(configurationName)

        End Sub

        Public Sub New(ByVal binding As System.ServiceModel.Channels.Binding, ByVal address As System.ServiceModel.EndpointAddress)

            MyBase.New(binding, address)

        End Sub

        Public Sub SendOrderResponse(ByVal msg As MsmqMessage(Of PurchaseOrder)) Implements IOrderResponse.SendOrderResponse

            MyBase.Channel.SendOrderResponse(msg)

        End Sub

    End Class

End Namespace
